include/aos/iam/nodeinfoprovider.hpp

```diff
@@ -49,6 +48,14 @@ static constexpr auto cAosComponentSM = "sm";
  */
 static constexpr auto cAosComponentUM = "um";
 
+/**
+ * Checks if the node is the main node.
+ *
+ * @param nodeInfo node info.
+ * @return RetWithError<bool>.
+ */
+RetWithError<bool> IsMainNode(const NodeInfo& nodeInfo);
```

Comment:
Shouldn't give any error, returning bool might be enough

Reply:
I'd keep it consistent with Find method of array